src/ln/functional_tests.rs

```diff
@@ -5697,6 +5697,168 @@ fn test_dynamic_spendable_outputs_local_htlc_success_tx() {
 	check_spends!(spend_txn[1], node_txn[2].clone());
 }
 
+#[test]
+fn test_fail_backwards_latest_remote_annouce() {
+	// Test that we fail backwards the full set of HTLCs we need to when remote announces its
+	// latest commitment transaction.
+	// This includes HTLCs which were below the dust threshold as well as HTLCs which were awaiting
+	// a remote RAA before they could be failed backwards (and combinations thereof).
+	// We also test duplicate-hash HTLCs by adding two nodes on each side of the target nodes which
+	// use the same payment hashes.
+	// Thus, we use a six-node network:
+	//
+	// A \         / E
+	//    - C - D -
+	// B /         \ F
+	// And test where C fails back to A/B when D announces its latest commitment transaction
+	let nodes = create_network(6);
+
+	create_announced_chan_between_nodes(&nodes, 0, 2);
+	create_announced_chan_between_nodes(&nodes, 1, 2);
+	let chan = create_announced_chan_between_nodes(&nodes, 2, 3);
+	create_announced_chan_between_nodes(&nodes, 3, 4);
+	create_announced_chan_between_nodes(&nodes, 3, 5);
+
+	// Rebalance and check output sanity...
+	send_payment(&nodes[0], &[&nodes[2], &nodes[3], &nodes[4]], 500000);
+	send_payment(&nodes[1], &[&nodes[2], &nodes[3], &nodes[5]], 500000);
+	assert_eq!(nodes[3].node.channel_state.lock().unwrap().by_id.get(&chan.2).unwrap().last_local_commitment_txn[0].output.len(), 2);
+
+	let ds_dust_limit = nodes[3].node.channel_state.lock().unwrap().by_id.get(&chan.2).unwrap().our_dust_limit_satoshis;
+	// 0th HTLC:
+	let (_, payment_hash_1) = route_payment(&nodes[0], &[&nodes[2], &nodes[3], &nodes[4]], ds_dust_limit*1000); // not added < dust limit + HTLC tx fee
+	// 1st HTLC:
+	let (_, payment_hash_2) = route_payment(&nodes[0], &[&nodes[2], &nodes[3], &nodes[4]], ds_dust_limit*1000); // not added < dust limit + HTLC tx fee
+	let route = nodes[1].router.get_route(&nodes[5].node.get_our_node_id(), None, &Vec::new(), ds_dust_limit*1000, TEST_FINAL_CLTV).unwrap();
+	// 2nd HTLC:
+	send_along_route_with_hash(&nodes[1], route.clone(), &[&nodes[2], &nodes[3], &nodes[5]], ds_dust_limit*1000, payment_hash_1);
+	// 3rd HTLC:
+	send_along_route_with_hash(&nodes[1], route, &[&nodes[2], &nodes[3], &nodes[5]], ds_dust_limit*1000, payment_hash_2);
+	// 4th HTLC:
+	let (_, payment_hash_3) = route_payment(&nodes[0], &[&nodes[2], &nodes[3], &nodes[4]], 1000000);
+	// 5th HTLC:
+	let (_, payment_hash_4) = route_payment(&nodes[0], &[&nodes[2], &nodes[3], &nodes[4]], 1000000);
+	let route = nodes[1].router.get_route(&nodes[5].node.get_our_node_id(), None, &Vec::new(), 1000000, TEST_FINAL_CLTV).unwrap();
+	// 6th HTLC:
+	send_along_route_with_hash(&nodes[1], route.clone(), &[&nodes[2], &nodes[3], &nodes[5]], 1000000, payment_hash_3);
+	// 7th HTLC:
+	send_along_route_with_hash(&nodes[1], route, &[&nodes[2], &nodes[3], &nodes[5]], 1000000, payment_hash_4);
```

Comment:
No, had a over the dust HTLC with same hash than a dust one to check that in case of a dust one being implicitly failed by a broadcast of commitment tx the over the dust isn't failed too.